src/main/java/org/jabref/logic/importer/fetcher/ISIDOREFetcher.java

```diff
@@ -0,0 +1,223 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Optional;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import org.jabref.logic.help.HelpFile;
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.IdBasedParserFetcher;
+import org.jabref.logic.importer.Parser;
+import org.jabref.logic.net.URLDownload;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.types.EntryType;
+import org.jabref.model.entry.types.StandardEntryType;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * Fetcher for ISIDORE (<a href="https://isidore.science">...</a>)
+ * Will take in the link to the website or the last six digits that identify the reference
+ * Uses ISIDORE's API.
+ * API explanation: <a href="https://isidore.science/api">...</a>
+ */
+public class ISIDOREFetcher implements IdBasedParserFetcher {
+    private static final int LINKLENGTH = 47;
+
+    private String URL;
+    private Parser parser;
+
+    public ISIDOREFetcher() {
+        this.parser = xmlData -> {
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            try {
+                DocumentBuilder builder = factory.newDocumentBuilder();
+                Document document = builder.parse(xmlData);
+
+                // Assuming the root element represents an entry
+                Element entryElement = document.getDocumentElement();
+
+                if (entryElement == null) {
+                    return Collections.emptyList();
+                }
+
+                return Collections.singletonList(xmlItemToBibEntry(document));
+            } catch (Exception e) {
+                // Handle parsing exceptions
+                try {
+                    throw new FetcherException("Issue with parsing link");
+                } catch (FetcherException ex) {
+                    throw new RuntimeException(ex);
+                }
```

Comment:
I am not sure why you map to a `RuntimeException` here.

The following code does not work?

```suggestion
                throw new FetcherException("Issue with parsing link");
```

If that does not work, try to use

```suggestion
    Unchecked.throwChecked(new FetcherException("Issue with parsing link"));
```

(This is from https://github.com/jOOQ/jOOL)

Reply:
Thank you, the first one causes an error but the second one works.